hydro_deploy/core/src/azure.rs

```diff
@@ -0,0 +1,570 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::{Context, Result};
+use async_ssh2_lite::{AsyncSession, SessionConfiguration};
+use async_trait::async_trait;
+use hydroflow_cli_integration::ServerBindConfig;
+use nanoid::nanoid;
+use serde_json::json;
+use tokio::net::TcpStream;
+
+use super::progress::ProgressTracker;
+use super::ssh::LaunchedSSHHost;
+use super::terraform::{TerraformOutput, TerraformProvider, TERRAFORM_ALPHABET};
+use super::util::async_retry;
+use super::{
+    ClientStrategy, Host, HostTargetType, LaunchedHost, ResourceBatch, ResourceResult,
+    ServerStrategy,
+};
+
+// #[derive(Debug)]
+pub struct LaunchedComputeEngine {
```

Comment:
Should probably be renamed to whatever Azure calls this (Compute Engine is a GCP term?). Also commented line above should be removed.